zeebe/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/AdHocSubProcessValidator.java

```diff
@@ -66,31 +64,37 @@ private static void validateTaskDefinition(
       return;
     }
 
-    final ZeebeAdHocImplementationType implementationType =
-        extensionElements.getChildElementsByType(ZeebeAdHoc.class).stream()
+    final ZeebeTaskDefinition taskDefinition =
+        extensionElements.getChildElementsByType(ZeebeTaskDefinition.class).stream()
             .findFirst()
-            .map(ZeebeAdHoc::getImplementationType)
-            .orElse(ZeebeAdHocImplementationType.BPMN);
+            .orElse(null);
 
-    final Collection<ZeebeTaskDefinition> taskDefinitions =
-        extensionElements.getChildElementsByType(ZeebeTaskDefinition.class);
+    if (taskDefinition == null) {
+      return;
+    }
 
-    if (implementationType == ZeebeAdHocImplementationType.BPMN && !taskDefinitions.isEmpty()) {
+    if (!adHocSubProcess.isCancelRemainingInstances()) {
       validationResultCollector.addError(
           0,
-          String.format(
-              "Must not have a zeebe:%s extension element with implementation type %s.",
-              ZeebeConstants.ELEMENT_TASK_DEFINITION, ZeebeAdHocImplementationType.BPMN));
+          "Must not set cancelRemainingInstances to false in combination with zeebe:taskDefinition");
```

Comment:
🔧 I think the error message is a bit odd. How about we say `Must not define cancelRemainingInstances in combination with zeebe:taskDefinition`? I know it's not technically correct, but if I'd see this message as user I'd change it to `true` and assume remaining instance will always get cancelled. With the message I proposed I'd remove it entirely.

Reply:
Good point, fully agree! 💯 

I updated the other 2 to have the same wording, so now all 3 return `Must not define ... in combination with zeebe:taskDefinition`.